src/server/handlers/launchpad.js

```diff
@@ -122,107 +143,137 @@ const responseError = (res, error) => {
+  const uri = `/repos/${owner}/${name}/contents/snapcraft.yaml`;
   const options = {
     headers: {
-      'Authorization': `token ${req.session.token}`,
+      'Authorization': `token ${token}`,
       'Accept': 'application/vnd.github.v3.raw'
     }
   };
-  logger.info(`Fetching snapcraft.yaml from ${repositoryUrl}`);
-  return requestGitHub.get(uri, options, (err, response, body) => {
-    if (response.statusCode !== 200) {
+  logger.info(`Fetching snapcraft.yaml from ${owner}/${name}`);
+  return requestGitHub.get(uri, options)
+    .then(checkStatus)
+    .then((response) => {
       try {
-        body = JSON.parse(body);
+        return yaml.safeLoad(response.body);
       } catch (e) {
-        logger.info('Invalid JSON received', err, body);
-        return res.status(500).send(RESPONSE_GITHUB_OTHER);
+        throw new PreparedError(400, RESPONSE_SNAPCRAFT_YAML_PARSE_FAILED);
       }
-      switch (body.message) {
-        case 'Not Found':
-          // snapcraft.yaml not found
-          return res.status(404).send(RESPONSE_GITHUB_NOT_FOUND);
-        case 'Bad credentials':
-          // Authentication failed
-          return res.status(401).send(RESPONSE_GITHUB_AUTHENTICATION_FAILED);
-        default:
-          // Something else
-          logger.info('GitHub API error:', err, body);
-          return res.status(500).send(RESPONSE_GITHUB_OTHER);
-      }
-    }
-
-    let snapcraftYaml;
-    try {
-      snapcraftYaml = yaml.safeLoad(body);
-    } catch (e) {
-      return res.status(400).send(RESPONSE_SNAPCRAFT_YAML_PARSE_FAILED);
-    }
-    return callback(snapcraftYaml);
-  });
+    });
 };
 
 export const newSnap = (req, res) => {
-  // XXX cjwatson 2016-12-15: Limit to only repositories the user owns.
-  if (!req.session || !req.session.token) {
-    return res.status(401).send(RESPONSE_NOT_LOGGED_IN);
-  }
-
-  getSnapcraftYaml(req, res, snapcraftYaml => {
-    const repositoryUrl = req.body.repository_url;
-    if (!('name' in snapcraftYaml)) {
-      return res.status(400).send(RESPONSE_SNAPCRAFT_YAML_NO_NAME);
-    }
-    const lp_client = getLaunchpad();
-    const username = conf.get('LP_API_USERNAME');
-    logger.info(`Creating new snap for ${repositoryUrl}`);
-    lp_client.named_post('/+snaps', 'new', {
-      parameters: {
-        owner: `/~${username}`,
-        distro_series: `/${DISTRIBUTION}/${DISTRO_SERIES}`,
-        name: `${makeSnapName(repositoryUrl)}-${DISTRO_SERIES}`,
-        git_repository_url: repositoryUrl,
-        git_path: 'refs/heads/master',
-        auto_build: true,
-        auto_build_archive: `/${DISTRIBUTION}/+archive/primary`,
-        auto_build_pocket: 'Updates',
-        processors: ARCHITECTURES.map(arch => {
-          return `/+processors/${arch}`;
-        }),
-        store_upload: true,
-        store_series: `/+snappy-series/${STORE_SERIES}`,
-        store_name: snapcraftYaml.name
+  const repositoryUrl = req.body.repository_url;
+  const lp_client = getLaunchpad();
```

Comment:
It already spread out a little bit in other places too as I noticed (I myself followed it in lauchpad tests I guess), but we should use camelCase consistently for JS variables.